homeassistant/components/sensor/__init__.py

```diff
@@ -337,6 +340,61 @@ def native_value(self) -> StateType | date | datetime | Decimal:
         """Return the value reported by the sensor."""
         return self._attr_native_value
 
+    @property
+    def native_precision(self) -> int | None:
+        """Return the number of digits after the decimal point for the sensor's state.
+
+        If native_precision is None, no rounding is done unless the sensor is subject
+        to unit conversion.
+
+        The display precision is influenced by unit conversion, a sensor which has
+        native_unit_of_measurement 'Wh' and is converted to 'kWh' will have its
+        native_precision increased by 3.
+        """
+        if hasattr(self, "_attr_native_precision"):
+            return self._attr_native_precision
+        if hasattr(self, "entity_description"):
+            return self.entity_description.native_precision
+        return None
+
+    @final
+    @property
+    def precision(self) -> int | None:
+        """Return the number of digits after the decimal point for the sensor's state.
+
+        This is the precision after unit conversion.
+        """
+        # Highest priority, for registered entities: precision set by user
+        if self._sensor_option_precision is not None:
+            return self._sensor_option_precision
+
+        # Second priority, native precision
+        if self.native_precision is None:
```

Comment:
No need to call the property method twice. As it is an integration implementation detail, it may actually involve data processing that is not needed twice.

```suggestion
        if precision := self.native_precision is None:
```